core/primitives/src/reed_solomon.rs

```diff
@@ -1,85 +1,51 @@
 use borsh::{BorshDeserialize, BorshSerialize};
 use itertools::Itertools;
-use reed_solomon_erasure::galois_8::{Field, ReedSolomon};
-use reed_solomon_erasure::ReconstructShard;
+use reed_solomon_erasure::galois_8::ReedSolomon;
 use std::io::Error;
 
-/// The ttl for a reed solomon instance to control memory usage. This number below corresponds to
-/// roughly 60MB of memory usage.
-const RS_TTL: u64 = 2 * 1024;
-
-/// Wrapper around reed solomon which occasionally resets the underlying
-/// reed solomon instead to work around the memory leak in reed solomon
-/// implementation <https://github.com/darrenldl/reed-solomon-erasure/issues/74>
-pub struct ReedSolomonWrapper {
-    rs: ReedSolomon,
-    ttl: u64,
+// Encode function takes a serializable object and returns a tuple of parts and length of encoded data
+pub fn rs_encode<T: BorshSerialize>(rs: &ReedSolomon, data: T) -> (Vec<Option<Box<[u8]>>>, usize) {
```

Comment:
nit: as these are public functions, I think we should expand the `rs_` prefix to help with readability.